web-ui/src/main/resources/catalog/components/userfeedback/GnUserfeedbackDirective.js

```diff
@@ -0,0 +1,411 @@
+                  tot = tot + scope.uf.ratingDATAQUALITY;
+                  i++;
+                }
+                if (scope.uf.ratingSERVICEQUALITY > 0) {
+                  tot = tot + scope.uf.ratingSERVICEQUALITY;
+                  i++;
+                }
+                if (scope.uf.ratingOTHER > 0) {
+                  tot = tot + scope.uf.ratingOTHER;
+                  i++;
+                }
+
+                if (tot > 0) {
+                  scope.uf.ratingAVG = Math.floor(tot / i);
+                } else {
+                  scope.uf.ratingAVG = 0;
+                }
+
+
+              };
+
+              function validateEmail(email) {
+                var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+                return re.test(email);
+              }
+
+              scope.submitForm = function(data, modal) {
+
+                if (!scope.loggedIn) {
+
+                  scope.authorNameError = false;
+                  scope.authorEmailError = false;
+                  scope.authorOrganizationError = false;
+
+                  if (!data.authorName) {
+                    scope.authorNameError = $translate.instant('GUFrequired');
+
+                    return false;
+                  }
+                  if (!data.authorEmail) {
+                    scope.authorEmailError = $translate.instant('GUFrequired');
+
+                    return false;
+                  }
+                  if (scope.uf.authorName.length > 64) {
+                    scope.authorNameError = $translate.instant('GUFtooLong');
+
+                    return false;
+                  }
+                  if (scope.uf.authorEmail.length > 64) {
+                    scope.authorEmailError = $translate.instant('GUFtooLong');
+
+                    return false;
+                  }
+
+                  var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+                  if (!re.test(scope.uf.authorEmail)) {
+                    scope.authorEmailError = $translate.instant('GUFnotValidFormat');
+
+                    return false;
+                  }
+                  if (scope.uf.authorOrganization.length > 64) {
+                    scope.authorOrganizationError = $translate.instant('GUFtooLong');
+
+                    return false;
+                  }
+                }
+
+                scope.uf.metadataUUID = scope.metatdataUUID;
+
+                if (angular.isUndefined(scope.metatdataUUID)) {
+                  console.log('Metadata UUID in null!');
+                  return;
+                }
+
+                $http.post('../api/userfeedback', data).success(function(data, status) {
+                  console.log(data);
+                  $window.location.reload();
```

Comment:
When user post the feedback, the page is fully reload - we don't need that. Only refresh the list of feedback no ?